cmd/podman/images/save.go

```diff
@@ -95,13 +98,24 @@ func save(cmd *cobra.Command, args []string) error {
 		if terminal.IsTerminal(int(fi.Fd())) {
 			return errors.Errorf("refusing to save to terminal. Use -o flag or redirect")
 		}
-		saveOpts.Output = "/dev/stdout"
+		pipePath, cleanup, err = setupPipe()
+		if err != nil {
+			return err
+		}
+		saveOpts.Output = pipePath
 	}
 	if err := parse.ValidateFileName(saveOpts.Output); err != nil {
 		return err
 	}
 	if len(args) > 1 {
 		tags = args[1:]
 	}
-	return registry.ImageEngine().Save(context.Background(), args[0], tags, saveOpts)
+	err = registry.ImageEngine().Save(context.Background(), args[0], tags, saveOpts)
+	if err != nil {
+		return err
+	}
+	if cleanup != nil {
+		defer cleanup()
```

Comment:
No, `defer` just after a successful `setupPipe()` - so that e.g. a `ValidateFileName` failure does not cause the command to exit without removing the temporary directory. And make sure the command does not hang in that case.